examples/mediadevices/src/index.js

```diff
@@ -120,23 +120,24 @@ function participantDisconnected(participant) {
 }
 
 // reads selected audio input, and updates preview and room to use the device.
-function applyAudioInputDeviceChange(event) {
+async function applyAudioInputDeviceChange(event) {
   var audio = document.querySelector('audio#audioinputpreview');
   var waveformContainer = document.querySelector('div#audioinputwaveform');
+  var deviceId = deviceSelections.audioinput.value;
   if (event) {
     event.preventDefault();
     event.stopPropagation();
   }
 
-  return applyAudioInputDeviceSelection(deviceSelections.audioinput.value, audio, someRoom).then(function() {
-    if (audio.srcObject) {
-      var canvas = waveformContainer.querySelector('canvas');
-      waveform.setStream(audio.srcObject);
-      if (!canvas) {
-        waveformContainer.appendChild(waveform.element);
-      }
+  await applyAudioInputDeviceSelection(deviceId, audio, someRoom);
```

Comment:
No need to declare a new variable if you are going to use it only once.
```suggestion
  await applyAudioInputDeviceSelection(deviceSelections.audioinput.value, audio, someRoom);
```